towerlib/towerlib.py

```diff
@@ -1908,7 +1907,15 @@ def create_job_template(self,  # pylint: disable=too-many-arguments, too-many-lo
         response = self.session.post(url, json=payload)
         if not response.ok:
             self._logger.error('Error creating job template, response was: "%s"', response.text)
-        return JobTemplate(self, response.json()) if response.ok else None
+            return None
+        job_template = JobTemplate(self, response.json())
+        if credential_id:
+            if not job_template.add_credential_by_id(credential_id):
+                self._logger.error('Error adding credentials, reverting the creation of the job template')
+                if not job_template.delete():
+                    raise InvalidJobTemplate(name)
```

Comment:
I think that this exception is misleading. Probably a new one would be more appropriate, like FailedToDeleteTemplate or something?